server/api/studio/pkg/ws/client.go

```diff
@@ -37,35 +39,49 @@ const (
 )
 
 var (
-	newline = []byte{'\n'}
-	space   = []byte{' '}
+	newline      = []byte{'\n'}
+	space        = []byte{' '}
+	nsidSpaceMap = utils.MutexMap[string]{}
 )
 
-var upgrader = websocket.Upgrader{
-	// ReadBufferSize:  1024,
-	// WriteBufferSize: 1024,
-	CheckOrigin: func(r *http.Request) bool {
-		return true
-	},
-	Error: func(w http.ResponseWriter, r *http.Request, status int, reason error) {
-		w.WriteHeader(status)
-		w.Write([]byte(reason.Error()))
-	},
-}
-
 // Client is a middleman between the websocket connection and the hub.
 type Client struct {
-	hub *Hub
-
+	hub        *Hub
 	clientInfo *auth.AuthData
-
 	// The websocket connection.
 	conn *websocket.Conn
-
 	// Buffered channel of outbound messages.
 	send chan []byte
 }
 
+func (c *Client) switchSpace(msgReceived *MessageReceive) *map[string]any {
+	reqSpace, ok := msgReceived.Body.Content["space"].(string)
+	currentSpace, _ := nsidSpaceMap.Get(c.clientInfo.NSID)
+
+	shouldSwitch := ok && reqSpace != "" && currentSpace != reqSpace
+	if !shouldSwitch {
+		return nil
+	}
```

Comment:
Why not `if !ok || reqSpace == "" ||  currentSpace == reqSpace { return nil}`?

Reply:
make it more semantic